File: Year-2023/LeetCode-March-Challenge/CanPlaceFlowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        for(int i = 0; i < flowerbed.length; ++i)
            if((flowerbed[i] == 0) && (i == 0 || flowerbed[i - 1] == 0) && (i == flowerbed.length - 1 || flowerbed[i + 1] == 0)){
                flowerbed[i] = 1;
                n--;
            }
        return n <= 0;
    }
}
public class CanPlaceFlowers{
    public static void main(String[] args) {
        int[] flowerbed = {1,0,0,0,0,1}; int n = 2;
        System.out.println(new Solution().canPlaceFlowers(flowerbed, n));
    }
}

File: Year-2023/LeetCode-March-Challenge/CanPlaceFlowers.js
var canPlaceFlowers = function(flowerbed, n) {
    for(let i = 0; i < flowerbed.length; ++i){
        if((flowerbed[i] === 0) && (i === 0 || flowerbed[i - 1] === 0) && (i === flowerbed.length - 1 || flowerbed[i + 1] === 0)){
            flowerbed[i] = 1;
            n--;
        }
    }
    return n <= 0;
};

let flowerbed = [1,0,0,0,0,1], n = 2;
console.log(canPlaceFlowers(flowerbed, n));